numpy/core/_add_newdocs_scalars.py

```diff
@@ -176,6 +176,46 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     Any Python object.
     """)
 
+add_newdoc_for_scalar_type('str_', ['str0', 'unicode_'],
+    r"""
+    A unicode string.
+
+    Unlike the builtin `str`, this supports the buffer protocol, exposing its
+    contents as UCS4:
+
+    >>> memoryview(np.str_("abcd"))
+    b'a\x00\x00\x00b\x00\x00\x00c\x00\x00\x00d\x00\x00\x00'
+
+    When used in array, this type cannot contain trailing nulls.
+    """)
+
+add_newdoc_for_scalar_type('bytes_', ['bytes0'],
+    r"""
+    A byte string.
+
+    When used in array, this type cannot contain trailing nulls.
+    """)
+
+add_newdoc_for_scalar_type('void', ['void0'],
+    """
+    Either an opaque sequence of bytes, or a structure.
```

Comment:
Any suggestions for a document to link to here?

Reply:
Perhaps a link to the structured array [docs](https://numpy.org/doc/stable/user/basics.rec.html)?

Reply:
Done, hopefully - can never remember if I need `:ref:`, the way `numpy_doc` handles `` `...` `` is a mystery to me.